pkgs/development/ocaml-modules/noise/default.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, buildDunePackage
+, fetchurl
+
+, benchmark
+, callipyge
+, chacha
+, digestif
+, hex
+, lwt
+, lwt_ppx
+, nocrypto
+, ounit
+, ppx_let
+, ppx_deriving
+, ppx_deriving_yojson
+, rfc7748
+}:
+
+buildDunePackage rec {
+  pname = "noise";
+  version = "0.2.0";
+
+  src = fetchurl {
+    url = "https://github.com/emillon/ocaml-noise/releases/download/v${version}/${pname}-v${version}.tbz";
+    sha256 = "sha256-fe3pT7fsuF2hCvXpInsRg6OvARs/eAh1Un454s1osDs=";
+  };
+
+  useDune2 = true;
+
+  minimumOCamlVersion = "4.05";
+
+  nativeBuildInputs = [
+    ppx_deriving
+    ppx_let
+  ];
+
+  propagatedBuildInputs = [
+    callipyge
+    chacha
+    digestif
+    lwt
+    nocrypto
+    rfc7748
```

Comment:
rfc7748 seems to be a recently introduced dependency which is not necessary for the 0.2.0 release: <https://github.com/emillon/ocaml-noise/blob/v0.2.0/noise.opam>